src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -61,7 +62,22 @@ public VisitResult(TypeWithState resultType, TypeSymbolWithAnnotations lValueTyp
             {
                 RValueType = resultType;
                 LValueType = lValueType;
+                // PROTOTYPE(nullable-api): Doesn't hold true for Tuple_Assignment_10. See if we can make it hold true
+                //Debug.Assert((RValueType.Type is null && LValueType.TypeSymbol is null) ||
+                //             RValueType.Type.Equals(LValueType.TypeSymbol, TypeCompareKind.ConsiderEverything | TypeCompareKind.AllIgnoreOptions));
+            }
+
+            public VisitResult(TypeSymbol type, NullableAnnotation annotation, NullableFlowState state)
+            {
+                RValueType = new TypeWithState(type, state);
+                LValueType = TypeSymbolWithAnnotations.Create(type, annotation);
+                Debug.Assert(RValueType.Type.Equals(LValueType.TypeSymbol, TypeCompareKind.ConsiderEverything));
             }
+
+            public VisitResult WithType(TypeSymbol newType) => new VisitResult(new TypeWithState(newType, RValueType.State), TypeSymbolWithAnnotations.Create(newType, LValueType.NullableAnnotation));
+            private string GetDebuggerDisplay() => $"{{LValue: {LValueType.GetDebuggerDisplay()}, RValue: {RValueType.GetDebuggerDisplay()}}}";
+
+            public static implicit operator NullabilityInfo(VisitResult result) => new NullabilityInfo(result.LValueType.NullableAnnotation.ToPublicAnnotation(), result.RValueType.State.ToPublicFlowState());
```

Comment:
>implicit operator [](start = 26, length = 17)

Where is this operator used? If it's only used in a small number of places, consider making this an explicit `ToNullabilityInfo()` method, and perhaps defined in the caller since the caller is presumably a consumer of `NullableWalker`.